modules/ztp-deploying-user-defined-firmware-configuration-with-gitops-ztp.adoc

```diff
@@ -0,0 +1,129 @@
+//
+// * edge_computing/ztp-deploying-far-edge-sites.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="ztp-deploying-user-defined-firmware-configuration-with-gitops-ztp_{context}"]
+= Deploying user-defined firmware to cluster hosts with {ztp}
+
+You can deploy user-defined firmware to cluster hosts by configuring the `SiteConfig` custom resource (CR) to include a hardware profile that you want to apply during cluster host provisioning.
+You can configure hardware profiles to apply to cluster hosts in the following scenarios:
+
+* All hosts in the cluster
+* Only hosts that meet certain criteria in the cluster
+* Individual hosts in the cluster
+
+.Prerequisites
+
+* You have installed the OpenShift CLI (`oc`).
+
+* You have installed {rh-rhacm-first} and logged in to the hub cluster as a user with `cluster-admin` privileges.
+
+* You have provisioned a cluster that is managed by {rh-rhacm}.
+
+* You created a Git repository where you manage your custom site configuration data.
+The repository must be accessible from the hub cluster and be defined as a source repository for the Argo CD application.
+
+.Procedure
+
+. Create the host firmware profile files that contain the firmware settings you want to apply at the site, cluster, and individual node level.
+For example:
++
+.site-firmware.profile
+[source,yaml]
+----
+BootMode: Uefi
+LogicalProc: Enabled
+ProcVirtualization: Enabled
+----
++
+.cluster-firmware.profile
+[source,yaml]
+----
+EnergyEfficientTurbo: Enabled
+ProcC1E: Disabled
+TpmSecurity: "On"
+----
++
+.node-firmware.profile
+[source,yaml]
+----
+EnergyEfficientTurbo: Disabled
+ProcTurboMode: Enabled
+ProcPwrPerf: OsDbpm
+ProcC1E: Disabled
+ProcCStates: Enabled
+----
+
+. Save the hardware profile YAML files relative to the `kustomization.yaml` file that you use to define how to provision the cluster, for example:
++
+[source,terminal]
+----
+example-ztp/install
+    └── site-install
+          ├── siteconfig-example.yaml
+          ├── kustomization.yaml
+          └── firmware-profiles
+                ├── site-firmware.profile
+                ├── cluster-firmware.profile
+                └── node-firmware.profile
+----
+
+. Edit the `SiteConfig` CR to include the firmware profiles that you want to apply in the cluster.
+For example:
++
+[source,yaml]
+----
+apiVersion: ran.openshift.io/v1
+kind: SiteConfig
+metadata:
+  name: "site-plan-cluster"
+  namespace: "example-cluster-namespace"
+spec:
+  baseDomain: "example.com"
+  # ...
+  biosConfigRef:
+    filePath: "./firmware-profiles/site-firmware.profile" <1>
+  clusters:
+    - clusterName: "cluster-1"
+      # ...
+      biosConfigRef:
+        filePath: "./firmware-profiles/cluster-firmware.profile" <2>
+      # ...
+      nodes:
+        - hostName: "compute-1.example.com"
+          # ...
+          bootMode: "UEFI"
+          biosConfigRef:
+            filePath: "./firmware-profiles/node-firmware.profile" <3>
+# ...
+----
+<1> Applies to all hosts deployed site-wide
```

Comment:
In the future we will support only one cluster per "SiteConfig". Maybe consider adding a note here like "(it is recommended to have only one cluster per SiteConfig CR)"

Reply:
Seems kind of strange that we are documenting doing site-wide configurations but suggesting to not do this...

I've added a note but not sure it works tbh.